src/adal5.service.ts

```diff
@@ -24,8 +23,9 @@ export class Adal5Service {
    * @memberOf Adal5Service
    */
   private adalContext: adal.AuthenticationContext;
-  private loginRefreshTimer : Subscription;
-
+  private loginRefreshTimer: Subscription;
+  private doRefresh: () => Promise<boolean>;
+  private doRefreshOption: () => Promise<boolean>;
```

Comment:
pas las peine de l'initialiser (redondant avec l.78), ou sinon au moins l'initialiser à `private doRefreshOption: () => this.doRefresh;`